### ISSUE TITLE
Set up debug Windows Trilinos nightly build

### ISSUE THREAD

**Issue by jwillenbring** [trilinos/Trilinos#1159]
@trilinos/framework 

This ticket is for setting up a nightly "debug" build for Trilinos on Windows.

Done for this story includes

-having a nightly build reporting to CDash with a debug configuration (with package options previously sent to Michael).

-the build should run on a Windows VM provided by the Trilinos team

-filing tickets for all errors found in executing the build. For all errors found, tickets should be filed in GitHub issues. If cause of error is known, a pull request could be offered to the package developers for package encountering the failure.

If necessary, another ticket will be filed for cleaning up all failures once the job is set up and running.

The initial work for setting up the build can be done anywhere, but getting the job running through Jenkins on the appropriate VM depends on tickets #1157 and #1158.

**Comment by mplooster** [comment]
When trying to build a debug version, I get the error 

> E:\TrilinosProject\Trilinos\packages\muelu\adapters\epetra\MueLu_CreateEpetraPreconditioner.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj

If this message is to be believed, I need to add the [/bigobj compiler flag](https://msdn.microsoft.com/en-us/library/ms173499.aspx) to get a debug build with Visual Studio 2015.

There are a couple of ways I can do this with CMake:

1. I can add the flag for specific files that need it using [set_source_file_properties](https://cmake.org/cmake/help/v3.2/command/set_source_files_properties.html). However, this approach might cause problems when new files are added.
2. I can add the flag for specific targets using [set_target_properties](https://cmake.org/cmake/help/v3.2/command/set_target_properties.html). (I'm not sure how well this one will work with TriBits.)
3. When building with MSVC, I can just add it to the list of compiler flags for everything.
a. I could do this as part of the Trilinos CMake files
b. I could do this as part of the build script, or as part of the call to CMake when configuring the build.
4. I can add the flag for MSVC debug AND release builds, or alternatively, just the debug build.

It doesn't seem like always having the flag there would be a bad thing, so option 3 and 4 (enabling it for everything) is looking pretty good to me. @trilinos/framework, what do you think? Do any of these options seem good to you, or perhaps you have other suggestions?

**Comment by bartlettroscoe** [comment]
Are there any downsides to using `/bigobj`?  The documentation:

* https://msdn.microsoft.com/en-us/library/ms173499.aspx

indicates the only disadvantage as "Linkers that shipped prior to Visual C++ 2005 cannot read .obj files that were produced with /bigobj."  I would say that is not an issues taht we should worry about.

I would say to go with "3. When building with MSVC, I can just add it to the list of compiler flags (CMAKE_CXX_FLAGS) for everything" and "3.b. do this as part of the Trilinos CMake files".  That way, you have no changes to individual packages and people trying to build Trilinos on Windows will not need to figure this out themselves.

**Comment by bmpersc** [comment]
I agree. Adding /bigobj automatically to the flags when building for Windows makes the most sense. With the increased use of templates in Trilinos I suspect that this will start to be a problem for other packages eventually so we might as well fix it for them now.

**Comment by mplooster** [comment]
Will do. Thanks for your input.

**Comment by bmpersc** [comment]
I have set the test that @mplooster had created to be run automatically through our jenkins server. It is currently set to run nightly and report to the experimental track.

There were a couple issues I had to address to get them to work with jenkins, but they were relatively minor, but one was a bit obscure. First the URL for the Trilinos repo was being set to the ssh version which required authentication tokens with github that I didn't want to set up for these builds since that shouldn't be necessary. I switched the URL back to the default https. This led to the obscure issue. The clone done by tribits wouldn't work with no reason given. It turns out the clone was timing out due to not being able to connect because of the proxies. I had to set the proxy information directly in the git configure, but once I did that it seemed to work and the rest of the builds worked mostly how I expected.

There are still some outstanding build and test errors, but we were expecting that as making every test clean was not part of the requirements. I am leaving the builds in the experimental track until we've either addressed the errors or have a plan that will get them addressed. At that point the builds will probably be moved to specialized, but the exact track can be determined at that time.

The build errors that are still outstanding look to be related to a limitation in windows. The error is:
fatal error C1083: Cannot open compiler generated file: '': Invalid argument
which after a bit of googling seems like it is probably related to the length of the filename being beyond the max filename length windows supports. This likely has a lot to do with the directory structure and will need to be investigated to see what can be done. I ran into this issue while installing compilers and tools so I'm not real surprised this happened. We will track that issue in a separate ticket since it isn't directly related to getting the testing setup.
